src/main/java/io/codeka/gaia/service/StackCostCalculator.java

```diff
@@ -0,0 +1,80 @@
+package io.codeka.gaia.service;
+
+import io.codeka.gaia.bo.JobStatus;
+import io.codeka.gaia.bo.JobType;
+import io.codeka.gaia.bo.Stack;
+import io.codeka.gaia.repository.JobRepository;
+import io.codeka.gaia.repository.TerraformModuleRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+/**
+ * This service calculates the cost of the running stack
+ */
+@Service
+public class StackCostCalculator {
+
+    private JobRepository jobRepository;
+
+    private TerraformModuleRepository moduleRepository;
+
+    @Autowired
+    public StackCostCalculator(JobRepository jobRepository, TerraformModuleRepository moduleRepository) {
+        this.jobRepository = jobRepository;
+        this.moduleRepository = moduleRepository;
+    }
+
+    /**
+     * Calculates an estimation of what this stack has cost in total
+     * @param stack
+     * @return
+     */
+    public BigDecimal calculateRunningCostEstimation(Stack stack){
+        var jobs = jobRepository.findAllByStackId(stack.getId());
+
+        if(jobs.isEmpty()){
+            return BigDecimal.ZERO;
+        }
+
+        var module = moduleRepository.findById(stack.getModuleId()).orElseThrow();
+
+        // calculate total running time
+        var duration = Duration.ofDays(0);
+
+        var start = LocalDateTime.now();
+        var end = LocalDateTime.now();
+
+        for(var job : jobs ){
+            // add
+            if(job.getType() == JobType.RUN && job.getStatus() == JobStatus.FINISHED){
+                start =job.getStartDateTime();
+                end = LocalDateTime.now();
+            }
+            if(job.getType() == JobType.STOP && job.getStatus() == JobStatus.FINISHED){
+                end = job.getStartDateTime();
+                duration = duration.plus(Duration.between(start, end));
+
+                // reset start and end
+                start = LocalDateTime.now();
+                end = LocalDateTime.now();
+            }
+        }
+
+        // add last duration
+        duration = duration.plus(Duration.between(start, end));
+
+        // get hourly cost
+        var hourlyCost = module.getEstimatedMonthlyCost().divide(BigDecimal.valueOf(31L*24L), 4, RoundingMode.HALF_UP);
```

Comment:
if the field is not filled, an exception is thrown when accessing the stack screen

Reply:
> I notice a difference in the render of the module description page as well.
> The padding which was previously on the readme part is no more.
> And the tabs look bigger too.
> 
> ![Screenshot_2019-08-09 Gaia - before](https://user-images.githubusercontent.com/12089967/62765320-2f1d0a00-ba90-11e9-9ca5-430c68947a81.png)
> 
> ![Screenshot_2019-08-09 Gaia - after](https://user-images.githubusercontent.com/12089967/62765319-2f1d0a00-ba90-11e9-8897-eb4fdcda1eb1.png)

I've corrected the size of the text, and added padding to the readme tab using the same <b-container fluid> as I did in the cost-ownership tab.

Reply:
Correction done.